File: src/main/java/ca/odell/glazedlists/impl/matchers/OrMatcher.java
/* Glazed Lists                                                 (c) 2003-2006 */
/* http://publicobject.com/glazedlists/                      publicobject.com,*/
/*                                                     O'Dell Engineering Ltd.*/
package ca.odell.glazedlists.impl.matchers;

import ca.odell.glazedlists.matchers.Matcher;

/**
 * A Matcher that matches if any child elements match.
 */
public class OrMatcher<E> implements Matcher<E> {

    /** The Matchers being combined with an "or" operator. */
    private final Matcher<? super E>[] matchers;

    @SafeVarargs
    public OrMatcher(Matcher<? super E>... matchers) {
        this.matchers = matchers;
    }

    /** {@inheritDoc} */
    @Override
    public boolean matches(E item) {
        for (Matcher<? super E> matcher : matchers) {
            if (matcher.matches(item))
                return true;
        }
        return false;
    }
}

File: src/main/kotlin/ca/odell/glazedlists/impl/matchers/OrMatcher.kt
/* Glazed Lists                                                 (c) 2003-2006 */
/* http://publicobject.com/glazedlists/                      publicobject.com,*/
/*                                                     O'Dell Engineering Ltd.*/
package ca.odell.glazedlists.impl.matchers

import ca.odell.glazedlists.matchers.Matcher

/** Matches when at least one child matcher matches. */
open class OrMatcher<E>(vararg matchers: Matcher<in E>) : Matcher<E> {
    private val matchers = matchers

    override fun matches(item: E): Boolean = matchers.any { it.matches(item) }
}
